worker/methods.go

```diff
@@ -238,6 +238,52 @@ func (w *Worker) TaskList() (tasks []TaskInfo, err error) {
 	return out["tasks"], nil
 }
 
+type TaskListParams struct {
+	CodeName string
+	Page     int
+	PerPage  int
+	FromTime time.Time
+	ToTime   time.Time
+	Statuses []string
+}
+
+func (w *Worker) FilteredTaskList(params TaskListParams) (tasks []TaskInfo, err error) {
+	out := map[string][]TaskInfo{}
+	url := w.tasks()
+
+	if params.CodeName != "" {
```

Comment:
I just don't want an empty parameter to be included in the request. That's why I check for emptyness. The request will be fired even if it's empty. I can still remove it though.

By the way, according to the documentation (http://dev.iron.io/worker/reference/api/#list_tasks), it's a required parameter.